templates/an-cash-dao/contracts/BaseTemplate.sol

```diff
@@ -0,0 +1,359 @@
+import "./lib/apps/Finance.sol";
+import "./lib/apps/TokenManager.sol";
+
+import "./lib/minime/MiniMeToken.sol";
+import "./lib/minime/MiniMeTokenFactory.sol";
+
+import "./lib/apm/ENS.sol";
+import "./lib/apm/Repo.sol";
+import "./lib/apm/PublicResolver.sol";
+
+import "./lib/os/ACL.sol";
+import "./lib/os/Kernel.sol";
+import "./lib/os/DAOFactory.sol";
+import "./lib/os/IsContract.sol";
+import "./lib/os/Uint256Helpers.sol";
+import "./lib/os/EVMScriptRegistry.sol";
+import "./lib/os/IFIFSResolvingRegistrar.sol";
+
+
+contract BaseTemplate is IsContract {
+    using Uint256Helpers for uint256;
+
+    address constant private ANY_ENTITY = address(-1);
+
+    // App IDs for [app].aragonpm.eth:
+    bytes32 constant internal AGENT_APP_ID = 0x9ac98dc5f995bf0211ed589ef022719d1487e5cb2bab505676f0d084c07cf89a;
+    bytes32 constant internal FINANCE_APP_ID = 0xbf8491150dafc5dcaee5b861414dca922de09ccffa344964ae167212e8c673ae;
+    bytes32 constant internal TOKEN_MANAGER_APP_ID = 0x6b20a3010614eeebf2138ccec99f028a61c811b3b1a3343b6ff635985c75c91f;
+    bytes32 constant internal VAULT_APP_ID = 0x7e852e0fcfce6551c13800f1e7476f982525c2b5277ba14b24339c68416336d1;
+
+    // App ID for agreement.precedence-campaign.aragonpm.eth:
+    bytes32 constant private AGREEMENT_APP_ID = 0x15a969a0e134d745b604fb43f699bb5c146424792084c198d53050c4d08126d1;
+    // App ID for disputable-voting.precedence-campaign.aragonpm.eth:
+    bytes32 constant private DISPUTABLE_VOTING_APP_ID = 0x39aa9e500efe56efda203714d12c78959ecbf71223162614ab5b56eaba014145;
+
+    string constant private ERROR_ENS_NOT_CONTRACT = "TEMPLATE_ENS_NOT_CONTRACT";
+    string constant private ERROR_DAO_FACTORY_NOT_CONTRACT = "TEMPLATE_DAO_FAC_NOT_CONTRACT";
+    string constant private ERROR_ARAGON_ID_NOT_PROVIDED = "TEMPLATE_ARAGON_ID_NOT_PROVIDED";
+    string constant private ERROR_ARAGON_ID_NOT_CONTRACT = "TEMPLATE_ARAGON_ID_NOT_CONTRACT";
+    string constant private ERROR_MINIME_FACTORY_NOT_PROVIDED = "TEMPLATE_MINIME_FAC_NOT_PROVIDED";
+    string constant private ERROR_MINIME_FACTORY_NOT_CONTRACT = "TEMPLATE_MINIME_FAC_NOT_CONTRACT";
+    string constant private ERROR_CANNOT_CAST_VALUE_TO_ADDRESS = "TEMPLATE_CANNOT_CAST_VALUE_TO_ADDRESS";
+    string constant private ERROR_INVALID_ID = "TEMPLATE_INVALID_ID";
+
+    ENS internal ens;
+    DAOFactory internal daoFactory;
+    MiniMeTokenFactory internal miniMeFactory;
+    IFIFSResolvingRegistrar internal aragonID;
+
+    event DeployDao(address dao);
+    event SetupDao(address dao);
+    event DeployToken(address token);
+    event InstalledApp(address appProxy, bytes32 appId);
+
+    constructor(DAOFactory _daoFactory, ENS _ens, MiniMeTokenFactory _miniMeFactory, IFIFSResolvingRegistrar _aragonID) public {
+        require(isContract(address(_ens)), ERROR_ENS_NOT_CONTRACT);
+        require(isContract(address(_daoFactory)), ERROR_DAO_FACTORY_NOT_CONTRACT);
+
+        ens = _ens;
+        aragonID = _aragonID;
```

Comment:
Maybe we can just inline the require checks for `aragonID` and `miniMeFactory` here, to remove the helper functions